sycl/source/detail/pi.cpp

```diff
@@ -684,22 +684,52 @@ DeviceBinaryImage::getProperty(const char *PropName) const {
   return *It;
 }
 
+// Returns the e_type field from an ELF image.
+static uint16_t getELFHeaderType(const unsigned char *ImgData, size_t ImgSize) {
+  bool IsBigEndian = ImgData[5] == 2;
+  if (IsBigEndian)
+    return (static_cast<uint16_t>(ImgData[16]) << 8) |
+           static_cast<uint16_t>(ImgData[17]);
+  uint16_t HdrType = 0;
+  std::copy(ImgData + 16, ImgData + 16 + sizeof(HdrType),
+            reinterpret_cast<char *>(&HdrType));
+  return HdrType;
+}
+
 RT::PiDeviceBinaryType getBinaryImageFormat(const unsigned char *ImgData,
                                             size_t ImgSize) {
   struct {
     RT::PiDeviceBinaryType Fmt;
     const uint32_t Magic;
   } Fmts[] = {{PI_DEVICE_BINARY_TYPE_SPIRV, 0x07230203},
-              {PI_DEVICE_BINARY_TYPE_LLVMIR_BITCODE, 0xDEC04342}};
+              {PI_DEVICE_BINARY_TYPE_LLVMIR_BITCODE, 0xDEC04342},
+              {PI_DEVICE_BINARY_TYPE_NATIVE, 0x43544E49}}; // INTC native
+
+  struct {
```

Comment:
Can we move this to line 722 please?

Reply:
Sure. It has been moved. It could potentially also be moved into the following check, though I further indentation may make it harder to read.